enhancements/sig-architecture/15-resourcebasedscheduling/README.md

```diff
@@ -0,0 +1,221 @@
+# Resource Based Scheduling
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management-io/website/)
+
+## Summary
+
+The proposed work would enhance the Placement APIs to support scheduling based on the resource usage/capacity of managed clusters.
+
+## Motivation
+
+During placement scheduling, managed clusters will be selected based on resource(s) (like memory and cpu etc.) usage/capacity to avoid scheduling workloads on clusters with insufficient resources.
+
+Link to the feature request: https://github.com/open-cluster-management-io/community/issues/52
+
+### User Stories
```

Comment:
is there a way to express: "I must have at least this much space"?  If not, why was that excluded?

Reply:
how does a user handle it when their clusters are autoscaling?  What do they expect to happen?  This could be presented as there not being capacity available I think.  Or perhaps that capacity is tight, but will expand.

Reply:
> is there a way to express: "I must have at least this much space"? If not, why was that excluded?

No, because we can not support it. Suppose we are able to create a placement which matches all clusters with at least 10G allocatable memory. In the first scheduling cycle, cluster1 is selected by this placement for it has 12G allocatable memory. After several minutes, the allocatable memory of cluster1 is reduced to 8G.  We don't know if the reduced 4G memory is consumed by workload associated with this placement. So in the next scheduling cycle, should cluster1 be selected by this placement or not?

Reply:
> how does a user handle it when their clusters are autoscaling? What do they expect to happen? This could be presented as there not being capacity available I think. Or perhaps that capacity is tight, but will expand.

Here is what will happens if managed clusters are autoscaling.
#### Scale up
1. User creates a placement with `ClusterResourcePreference`;
2. The cluster with most allocatable to capacity ratio or most allocatable will be selected;
3. Workload will then be deployed on the selected cluster;
4. If there is no enough resource available or the capacity is tight, new node will be added;
5. Capacity/allocatable of this cluster changes. That will trigger a new scheduling cycle for placements with ClusterResourcePreference;

#### Scale down
1. An underutilized node in a managed cluster is removed;
2. Capacity/allocatable of this cluster changes. That will trigger a new scheduling cycle for placements with ClusterResourcePreference;
3. Some Placements may no longer select this cluster because of resource change;